### PR TITLE
Fix for obsolete URI.unescape

### PR DESCRIPTION
**Pull request by tatsianaclifton**
Changes obsolete URI.unescape method to URI.decode_www_form_component

Ruby docs https://ruby-doc.org/stdlib-2.7.5/libdoc/uri/rdoc/URI/Escape.html

ECS-2123

Unit tests pass the same as on master (failed tests are not related to `Riak::Util::Escape` class)
Finished in 0.70327 seconds (files took 0.74945 seconds to load)
628 examples, 3 failures

### CODE REVIEW

File: lib/riak/util/escape.rb [hunk 1]
```diff
@@ -87,7 +87,11 @@ def maybe_unescape(bucket_or_key)
       # @param [String] bucket_or_key the bucket or key name
       # @return [String] the unescaped name
       def unescape(bucket_or_key)
-        Riak.escaper.unescape(bucket_or_key.to_s)
+        if Riak.escaper == URI
+          Riak.escaper.decode_www_form_component(bucket_or_key.to_s)
```
Comment (aenand) [inline lib/riak/util/escape.rb:91]: Would we want to use `CGI.unescape` instead of `decode_www_form_component` since core and AM both use `CGI.unescape`?

Comment (tatsianaclifton) [reply lib/riak/util/escape.rb:91]: This module supports 2 types of escapers - `URI` and `CGI`. If it is `CGI` it is in `else` part where `CGI.unescape` is used. The `decode_www_form_component` method is for cases when `URI` escaper is used what is checked on line 90. The application that uses this riak library can set escaper I assume for what is needed.

**Comment by aenand** [comment]
Is there any way to test this (other than running unit tests) @tatsianaclifton?

**Comment by tatsianaclifton** [comment]
> Is there any way to test this (other than running unit tests) @tatsianaclifton?

Unfortunately, I'm not an expert for that gem, so it is hard for me to figure out exactly how and where this method is used. So I would think besides unit tests testing can be done by getting that changes into the application and ensure application still works as intended. But I really don't know what area to look closer :confused:

**Review by aenand — APPROVED** [review]
The changes look good. We'll have to monitor the deploy carefully 🚀

**Comment by tatsianaclifton** [comment]
@aenand I also changed obsolete `URI.escape` method to `URI.encode_www_form_component` if you can take a look as I made this change after you approval. The encode_www_form_component is a bit different from URI.escape, URI.escape encodes only unsafe characters when encode_www_form_component encodes all as percent-encoded octets and space as a `+`. So with `URI.escape` chars such as [ , ], @ won't be encoded with `URI.encode_www_form_component` they will. So. I had to change unit tests for that. I think this difference should be not an issue, because according URI standard it seems that encoding reserved chars is recommended https://datatracker.ietf.org/doc/html/rfc3986#section-2.2. Additionally, we probably don't use any chars in bucket and key names in riak. Also, decoding method `URI.decode_www_form_component` decodes everything that was encoded with `URI.encode_www_form_component` correctly.

**Comment by aenand** [comment]
@tatsianaclifton that change makes a lot of sense since we are using `decode_www_form_component ` to decode. Seems good to me!

**Comment by tatsianaclifton** [comment]
> @tatsianaclifton that change makes a lot of sense since we are using `decode_www_form_component ` to decode. Seems good to me!

Thanks for re-reviewing

**Pull request merged**